DQM/CTPPS/plugins/TotemT2DQMSource.cc

```diff
@@ -0,0 +1,281 @@
+  void bookHistograms(DQMStore::IBooker&, const edm::Run&, const edm::EventSetup&) override;
+  void analyze(const edm::Event&, const edm::EventSetup&) override;
+
+private:
+  void fillActivePlanes(std::unordered_map<unsigned int, std::set<unsigned int>>&, const TotemT2DetId&);
+  void fillTriggerBitset(const TotemT2DetId&);
+  void clearTriggerBitset();
+  bool areChannelsTriggered(const TotemT2DetId&);
+  void bookErrorFlagsHistogram(DQMStore::IBooker&);
+  void fillErrorFlagsHistogram(const TotemT2Digi&);
+  void fillEdges(const TotemT2Digi&, const TotemT2DetId&);
+  void fillToT(const TotemT2RecHit&, const TotemT2DetId&);
+
+  edm::ESGetToken<TotemGeometry, TotemGeometryRcd> geometryToken_;
+  edm::EDGetTokenT<edmNew::DetSetVector<TotemT2Digi>> digiToken_;
+  edm::EDGetTokenT<edmNew::DetSetVector<TotemT2RecHit>> rechitToken_;
+
+  std::unique_ptr<TotemT2Segmentation> segm_;
+
+  double HPTDC_BIN_WIDTH_NS_ = 25. / 1024;
+  MonitorElement* HPTDCErrorFlags_2D_ = nullptr;
+
+  unsigned int nbinsx_, nbinsy_;
+  unsigned int windowsNum_;
+
+  struct SectorPlots {
+    MonitorElement* activePlanes = nullptr;
+    MonitorElement* activePlanesCount = nullptr;
+
+    MonitorElement* triggerEmulator = nullptr;
+    std::bitset<(TotemT2DetId::maxPlane + 1) * (TotemT2DetId::maxChannel + 1)> hitTilesArray;
+    static const unsigned int MINIMAL_TRIGGER = 3;
+
+    MonitorElement *leadingEdge = nullptr, *trailingEdge = nullptr, *timeOverTreshold = nullptr;
+
+    SectorPlots() = default;
+    SectorPlots(
+        DQMStore::IBooker& ibooker, unsigned int id, unsigned int nbinsx, unsigned int nbinsy, unsigned int windowsNum);
+  };
+
+  struct PlanePlots {
+    MonitorElement* digisMultiplicity = nullptr;
+    MonitorElement* rechitMultiplicity = nullptr;
+
+    PlanePlots() = default;
+    PlanePlots(DQMStore::IBooker& ibooker, unsigned int id, unsigned int nbinsx, unsigned int nbinsy);
+  };
+
+  std::unordered_map<unsigned int, SectorPlots> sectorPlots_;
+  std::unordered_map<unsigned int, PlanePlots> planePlots_;
+};
+
+TotemT2DQMSource::SectorPlots::SectorPlots(
+    DQMStore::IBooker& ibooker, unsigned int id, unsigned int nbinsx, unsigned int nbinsy, unsigned int windowsNum) {
+  std::string title, path;
+
+  TotemT2DetId(id).armName(path, TotemT2DetId::nPath);
+  ibooker.setCurrentFolder(path);
+
+  TotemT2DetId(id).armName(title, TotemT2DetId::nFull);
+
+  activePlanes = ibooker.book1D("active planes", title + " which planes are active;plane number", 8, -0.5, 7.5);
+
+  activePlanesCount = ibooker.book1D(
+      "number of active planes", title + " how many planes are active;number of active planes", 9, -0.5, 8.5);
+
+  triggerEmulator = ibooker.book2DD("trigger emulator",
+                                    title + " trigger emulator;arbitrary unit;arbitrary unit",
+                                    nbinsx,
+                                    0.,
+                                    nbinsx,
```

Comment:
This parameter should be a `double`
Then, for some reasonable histogram that reports integers in nbinsx bins, I imagine that it should better go from `-0.5` to `double(nbinsx)-0.5` (same for `nbinsy`)